### PR TITLE
Added GitLab integration example to docs

### PR DESCRIPTION
**Pull request by brandonSc**
Full demo on GitLab running here: https://gitlab.com/brandon176/earthly-demo/-/tree/main

### CODE REVIEW

**Comment by brandonSc** [comment]
I'm not that familiar with GitBook - is there a way to "build" the docs locally to ensure the new page is working?

**Comment by alexcb** [comment]
we'll need to add a link to this guide under https://github.com/earthly/earthly/blob/main/docs/SUMMARY.md?plain=1#L55

File: docs/ci-integration/guides/gitlab-integration.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    - /usr/local/bin/earthly bootstrap
+    - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY
+
+earthly:
+  stage: build
+  script:
+    - earthly --ci --push -P +build
+```
+
+For the full example on GitLab, see the [demo project here](https://gitlab.com/brandon176/earthly-demo/-/tree/main).
```
Comment (alexcb) [inline docs/ci-integration/guides/gitlab-integration.md:26]: NIT: maybe we can avoid using the word `here` in the link, by wording it like:
```suggestion
A full example is available [on GitLab](https://gitlab.com/brandon176/earthly-demo/-/tree/main).
```

File: docs/ci-integration/guides/gitlab-integration.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+
+# GitLab CI/CD integration
+
+Here is an example a GitLab CI/CD build, where we build the Earthly target `+build`.
```
Comment (alexcb) [inline docs/ci-integration/guides/gitlab-integration.md:4]: NIT: avoid starting with "Here is"; maybe we can state it more concisely: 
```suggestion
This example uses [GitLab CI/CD](https://docs.gitlab.com/ee/ci/) to build the Earthly target `+build`.

```

**Comment by alexcb** [comment]
Nice work on the gitlab example project! Just a couple minor nitpicky comments.

**Comment by brandonSc** [comment]
> Nice work on the gitlab example project! Just a couple minor nitpicky comments.

Thanks, and good suggestions!

**Review by alexcb — APPROVED** [review]

File: docs/ci-integration/guides/gitlab-integration.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    - /usr/local/bin/earthly bootstrap
+    - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY
+
+earthly:
+  stage: build
+  script:
+    - earthly --ci --push -P +build
+```
+
+A full example is available [on GitLab](https://gitlab.com/brandon176/earthly-demo/-/tree/main).
```
Comment (vladaionescu) [inline docs/ci-integration/guides/gitlab-integration.md:27]: Should this be under an `earthly` GitLab user perhaps?

File: docs/ci-integration/guides/gitlab-integration.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    - apk update && apk add git
+    - wget https://github.com/earthly/earthly/releases/download/v0.6.1/earthly-linux-amd64 -O /usr/local/bin/earthly
+    - chmod +x /usr/local/bin/earthly
+    - /usr/local/bin/earthly bootstrap
+    - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY
+
+earthly:
+  stage: build
+  script:
+    - earthly --ci --push -P +build
```
Comment (vladaionescu) [inline docs/ci-integration/guides/gitlab-integration.md:24]: Seems that the output is not colorized. Perhaps it's worth adding `FORCE_COLOR=1` in the env.

File: docs/ci-integration/guides/gitlab-integration.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+```yml
+# .gitlab-ci.yml
+
+image: docker
+services:
+  - docker:dind
+
+before_script:
+    - apk update && apk add git
+    - wget https://github.com/earthly/earthly/releases/download/v0.6.1/earthly-linux-amd64 -O /usr/local/bin/earthly
```
Comment (vladaionescu) [inline docs/ci-integration/guides/gitlab-integration.md:16]: Seems that this is one version behind - mind updating to `v0.6.2`? Can you also check `release/README.md` to see if the process to update versions in docs would cover this page too?

**Review by vladaionescu — COMMENTED** [review]
🎉 

Got some minor comments below...

**Pull request merged**